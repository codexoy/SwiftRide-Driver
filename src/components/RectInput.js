import React from "react"
import {
	TouchableOpacity,
	View,
	StyleSheet,
	TextInput,
	Text,
	Switch
} from "react-native"
import { Icon } from "react-native-elements"
import { colors } from "../consts"

export default class RectInput extends React.Component {
	constructor(props) {
		super(props)
		this.defaultColor = "#adadad"
		this.state = {
			color: this.defaultColor
		}
	}
	onPress = () => {
		if (this.input) this.input.focus()
		else if (this.props.hasOwnProperty("onPress")) {
			this.props.onPress()
		} else {
			// this is a switch, toggle it
			this.props.onChangeText(!this.props.value)
		}
	}
	onFocus = () => {
		this.setState({
			color: colors.blue
		})
		if (this.props.hasOwnProperty("onFocus")) this.props.onFocus()
	}
	onBlur = () => {
		this.setState({
			color: this.defaultColor
		})
		if (this.props.hasOwnProperty("onBlur")) this.props.onBlur()
	}
	render() {
		let iconType = "material"
		if (this.props.iconType) iconType = this.props.iconType
		let input,
			leftSide,
			iconStyle,
			labelStyle = {}
		if (this.props.empty) {
			input = <View />
			leftSide = this.props.leftSide
			iconStyle = { marginBottom: 6 }
			labelStyle = { marginTop: 6 }
		} else if (!this.props.switch) {
			input = (
				<TextInput
					value={this.props.value}
					testID={this.props.testID}
					style={{
						...styles.input,
						...this.props.style
					}}
					autoFocus={this.props.autoFocus || false}
					ref={input => {
						this.input = input
					}}
					onSubmitEditing={this.props.onSubmitEditing}
					onFocus={this.onFocus.bind(this)}
					onBlur={this.onBlur.bind(this)}
					onChangeText={this.props.onChangeText}
					secureTextEntry={this.props.secureTextEntry || false}
				/>
			)
			leftSide = (
				<Icon
					type={iconType}
					name={this.props.iconName}
					color={this.state.color}
				/>
			)
			iconStyle = { marginTop: 8 }
		} else {
			input = <View />
			iconStyle = { marginBottom: 6 }
			leftSide = (
				<Switch
					value={this.props.value}
					onValueChange={this.props.onChangeText}
				/>
			)
			labelStyle = { marginTop: 6 }
		}
		return (
			<TouchableOpacity
				style={styles.inputView}
				onPress={this.onPress.bind(this)}
				activeOpacity={1}
			>
				<View style={styles.row}>
					<View style={styles.column}>
						<Text
							style={{
								...styles.inputLabel,
								...labelStyle,
								color: this.state.color
							}}
						>
							{this.props.label}
						</Text>
						{input}
					</View>
					<View style={{ ...styles.iconView, ...iconStyle }}>
						{leftSide}
					</View>
				</View>
			</TouchableOpacity>
		)
	}
}

const styles = StyleSheet.create({
	inputView: {
		width: "100%",
		borderBottomWidth: 1,
		borderBottomColor: "#f7f7f7",
		paddingBottom: 6,
		alignItems: "flex-start",
		paddingHorizontal: 20,
		paddingVertical: 12
	},
	inputLabel: {
		color: "#adadad",
		fontWeight: "bold",
		fontSize: 14
	},
	column: {
		alignItems: "flex-start",
		flex: 1,
		flexDirection: "column"
	},
	input: {
		padding: 6,
		textAlign: "right"
	},
	iconView: {
		marginLeft: "auto"
	},
	row: {
		flexDirection: "row"
	}
})
