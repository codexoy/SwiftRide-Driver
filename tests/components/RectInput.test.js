import { Text } from "react-native"
import React from "react"

import renderer from "react-test-renderer"

import RectInput from "../../src/components/RectInput"

describe("RectInput", () => {
	test("view renders correctly", () => {
		const func = jest.fn()
		const tree = renderer
			.create(
				<RectInput
					name={"test"}
					autoFocus={false}
					onFocus={func}
					onBlur={func}
					onChangeText={func}
					iconName="done"
					iconType="material"
					onSubmitEditing={func}
					style={{ marginTop: 20 }}
				/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
