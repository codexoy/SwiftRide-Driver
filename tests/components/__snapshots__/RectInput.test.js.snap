// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RectInput view renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "borderBottomColor": "#f7f7f7",
      "borderBottomWidth": 1,
      "opacity": 1,
      "paddingBottom": 6,
      "paddingHorizontal": 20,
      "paddingVertical": 12,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#adadad",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoFocus={false}
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        onFocus={[Function]}
        onSubmitEditing={[MockFunction]}
        secureTextEntry={false}
        style={
          Object {
            "marginTop": 20,
            "padding": 6,
            "textAlign": "right",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "marginLeft": "auto",
          "marginTop": 8,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          underlayColor="white"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#adadad",
                  "fontSize": 24,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            î¡¶
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`RectInput view renders correctly with empty=true 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "borderBottomColor": "#f7f7f7",
      "borderBottomWidth": 1,
      "opacity": 1,
      "paddingBottom": 6,
      "paddingHorizontal": 20,
      "paddingVertical": 12,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#adadad",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginTop": 6,
          }
        }
      />
      <View />
    </View>
    <View
      style={
        Object {
          "marginBottom": 6,
          "marginLeft": "auto",
        }
      }
    >
      <Text>
        test
      </Text>
    </View>
  </View>
</View>
`;

exports[`RectInput view renders correctly with switch=true 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "borderBottomColor": "#f7f7f7",
      "borderBottomWidth": 1,
      "opacity": 1,
      "paddingBottom": 6,
      "paddingHorizontal": 20,
      "paddingVertical": 12,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#adadad",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginTop": 6,
          }
        }
      />
      <View />
    </View>
    <View
      style={
        Object {
          "marginBottom": 6,
          "marginLeft": "auto",
        }
      }
    >
      <RCTSwitch
        accessibilityRole="button"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 31,
            "width": 51,
          }
        }
        value={false}
      />
    </View>
  </View>
</View>
`;
